### ISSUE TITLE
Use git-submodules to include UtilityToken and Organized contracts

### ISSUE THREAD

**Issue by deepesh-kn** [OpenST/mosaic-contracts#712]
**Use git-submodules to include the following contracts** 
- SafeMath.sol
- EIP20Token.sol
- UtilityToken.sol 
- Organized.sol

**Following is the proposal:**

1. Remove the existing implementation of the above contracts.
2. Use git-submodules to include EIP20Token.sol and UtilityToken.sol from `OpenST/utilitytoken-contracts`.
3. Check if UtilityToken imports the dependent Organized contract.	
4. If not then use `git-submodules` to include Organized contract in the project from `OpenST/organization-contracts`.
5. Use git-submodules to include SafeMath contract in the project from < will be updated >

**Comment by jasonklein** [comment]
@deepesh-kn: in addition to git-submodules, another option might be truffle's existing support for NPM packages (I have not yet tried it, so this is a suggestion and not a recommendation 😅). Have you considered that option?

**Comment by schemar** [comment]
> @deepesh-kn: in addition to git-submodules, another option might be truffle's existing support for [NPM packages](https://truffleframework.com/docs/truffle/getting-started/package-management-via-npm) (I have not yet tried it, so this is a suggestion and not a recommendation 😅). Have you considered that option?
> 
> ✏️ N.B.: I will also post this question on the epic.

Now discussed here: https://github.com/OpenST/developer-guidelines/issues/20#issuecomment-483185693

**Comment by benjaminbollen** [comment]
EIP20token should not be in utility token.  if we want to deduplicate it then we can use openZeppeling eg. but for now EIP20token is not part of this scope.  It should only ever be part of test contracts

**Comment by schemar** [comment]
Blocked by the discussion git submodules vs npm.
EIP20 should come from openzeppelin.

**Comment by benjaminbollen** [comment]
conclusion from discussion;

internal dependencies like organization-contracts and utilitytoken-contracts should be included through git-submodule;

external dependencies like SafeMath (possibly an EIP20Token for testing purposes) should be included through NPM packages

**Comment by schemar** [comment]
* SafeMath and EIP20 should be imported using NPM
* OST dependencies should be imported using git-submodules

See also https://github.com/OpenST/openst-contracts/issues/197

**Comment by schemar** [comment]
There are conflicts due to multiple declarations of the same contracts. For example, organization contracts are part of utility tokens; safe math from utility tokens is already declared in npm dependency ...

Using NPM would have automatically deduplicated the contracts.

**Comment by schemar** [comment]
I think step one to resolving this is adding safemath as an npm dependency to utilitytoken contracts.

**Comment by schemar** [comment]
Step two will be using the organization contracts from within utility token contracts.

"inception"

**Comment by schemar** [comment]
Deduplicating EIP20Token is out-of-scope, as it requires changing UtilityToken contracts (which has an EIP20Token) and all consumers of UtilityToken, which currently may or may not rely on the EIP20Token from there.
Simply adding the npm package does not work, as there will be an error about multiple definitions of the same Symbol (`contract EIP20Token`).
